### PR TITLE
feat(batch): enable B>1 DiT forward in SP path (1.42× at B=2, B=1 bit-identical)

### PR DESCRIPTION
**Pull request by prashant182**
## Summary

Three minimal changes to `sp_dit_forward_causal` that unlock batched multi-user inference. Each is a no-op at B=1; together they let B>1 run correctly through the same path.

**B=1 bit-identical** (MD5 `ed2f82628308a3f8acd9b7935bb84401`). **B=2 throughput: 1.42×** (single-chunk forward, two distinct users batched).

## What changed

| Site | Fix | Why no-op at B=1 |
|---|---|---|
| `assert len(x) == 1` | removed | Assertion was always true at B=1; surrounding code already handles a list. |
| `torch.cat(c2ws, dim=1)` → `dim=0` | each user keeps its own camera conditioning | One-element list — both dims give the same tensor. |
| `t.expand(B, S)` → `t.unsqueeze(1).expand(B, S)` | works for any B | At B=1 the broadcast result is identical. |

## Verification

```
B=1 bench:    MD5 ed2f82628308a3f8acd9b7935bb84401 (locked baseline)
              generate() 11690 ms (noise vs prior)

B=2 probe:    user A match (atol=3e-2): PASS
              user B match (atol=3e-2): PASS
              throughput 1.42× (272 ms for 2 users vs 388 ms sequential)
```

Tolerance is set at bf16 attention precision (flash-attn reduces matmuls in different orders for B=1 vs B=2 → tiny numerical drift, max abs diff 1.8e-2 / mean 2.8e-3 across ~1M elements).

## Scope

Lifts the inner DiT forward only. The higher-level `generate()` loop still constructs B=1 inputs, so production paths are unchanged. Multi-user serving infrastructure (per-user session, continuous-batching scheduler, PagedAttention for heterogeneous timelines) is separate follow-on work.

## Stack

Stacks on #51, #52, #53. Reported diff shrinks once those land.

### CODE REVIEW

**Comment by prashant182** [comment]
Nudge for review when you have a moment Three minimal changes in `sp_dit_forward_causal` that lift the B=1 lock. Each is a no-op at B=1 (verified: MD5 unchanged on the locked baseline), and together they let B=2 run correctly. **1.42× throughput at B=2**, per-batch outputs match B=1 references within bf16 precision. This is the multi-tenant primitive working end to end on top of #53.

cc @robbyant @JingyeChen

**Comment by prashant182** [comment]
Do you guys have the slack or some other realtime communication tool somewhere to shorten the contribution loop let me know and I can join that forum.

**Comment by prashant182** [comment]
Hi checking to see if you can review these PRs please? @qiuyu96
